knowledge-base/editor-add-text-templates.md

```diff
@@ -0,0 +1,112 @@
+---
+title: Add text templates in Editor
+description: How to insert text templates in Editor?
+type: how-to
+page_title: Add text templates in Editor
+slug: editor-kb-add-text-templates
+position: 
+tags: editor, text, template, insert, add
+ticketid: 1589596
+res_type: kb
+---
+
+## Environment
+<table>
+	<tbody>
+		<tr>
+			<td>Product</td>
+			<td>Editor for Blazor</td>
+		</tr>
+	</tbody>
+</table>
+
+
+## Description
+
+Is there a way to insert templated text in the Editor?
+
+I want to have a tool that prefills various templates in the Editor. How to achieve this?
+
+## Solution
+
+You can achieve the desired result by creating a custom tool that programmatically changes the Editor value (inserts the desired templated text in the Editor).
+
+1. [Create custom tool]({%slug editor-custom-tools%}). If you want to include more than one template option, you may consider a [DropDownList]({%slug components/dropdownlist/overview%}) component as per the example below. Thus, the user will be able to easily select one of the suggested templates.
+
+1. Provide a colection of your templates to the `Data` parameter of the DropDownList. The example below shows how you can use a complex model for the templates. The DropDownList only uses an `Id` and `Title` to show the available templates to the user. Once the user selects the desired template, you may retrieve its content from the database, for example. The key part is that the content of the template should be provided to the Editor as `string`, so the component can render it.
+
+1. Handle the [`ValueChanged`]({%slug components/dropdownlist/events%}#valuechanged) event of the DropDownList to get the template value and update the Editor value. You may add the template value to the Editor or replace the whole content depending on the exact desired result.
```

Comment:
It may be better to not apply the selected value to the DropDownList. Currently, the user has no easy way to add the same template two times.